api/v1beta1/cinder_types.go

```diff
@@ -17,57 +17,148 @@ limitations under the License.
 	CinderAPIContainerImage string `json:"cinderAPIContainerImage,omitempty"`
-	// Cinder Backup Container Image URL
+
+	// +kubebuilder:validation:Optional
+	// CinderBackupContainerImage - Cinder Backup Container Image URL
 	CinderBackupContainerImage string `json:"cinderBackupContainerImage,omitempty"`
-	// Cinder Scheduler Container Image URL
+
+	// +kubebuilder:validation:Optional
+	// CinderSchedulerContainerImage - Cinder Scheduler Container Image URL
 	CinderSchedulerContainerImage string `json:"cinderSchedulerContainerImage,omitempty"`
-	// Cinder API Replicas
+
+	// +kubebuilder:validation:Required
+	// CinderAPIReplicas - Cinder API Replicas
 	CinderAPIReplicas int32 `json:"cinderAPIReplicas"`
-	// Cinder Backup Replicas
+
+	// +kubebuilder:validation:Required
+	// CinderBackupReplicas - Cinder Backup Replicas
 	CinderBackupReplicas int32 `json:"cinderBackupReplicas"`
-	// Cinder Backup node selector
-	CinderBackupNodeSelectorRoleName string `json:"cinderBackupNodeSelectorRoleName,omitempty"`
-	// Cinder Scheduler Replicas
+
+	// +kubebuilder:validation:Required
+	// CinderSchedulerReplicas - Cinder Scheduler Replicas
 	CinderSchedulerReplicas int32 `json:"cinderSchedulerReplicas"`
-	// Secret containing: CinderPassword, TransportURL
-	CinderSecret string `json:"cinderSecret,omitempty"`
-	// Secret containing: NovaPassword
-	NovaSecret string `json:"novaSecret,omitempty"`
-	// Cells to create
+
+	// +kubebuilder:validation:Optional
+	// CinderVolumes - cells to create
 	CinderVolumes []Volume `json:"cinderVolumes,omitempty"`
 }
 
 // Volume defines cinder volume configuration parameters
 type Volume struct {
 	// Name of cinder volume service
 	Name string `json:"name,omitempty"`
+
+	// +kubebuilder:validation:Optional
 	// Cinder Volume Container Image URL
-	CinderVolumeContainerImage string `json:"cinderVolumeContainerImage,omitempty"`
+	ContainerImage string `json:"containerImage,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default=1
 	// Cinder Volume Replicas
-	CinderVolumeReplicas int32 `json:"cinderVolumeReplicas"`
+	Replicas int32 `json:"replicas"`
+
 	// Cinder Volume node selector
-	CinderVolumeNodeSelectorRoleName string `json:"cinderVolumeNodeSelectorRoleName,omitempty"`
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="# add your customization here"
+	// CustomServiceConfig - customize the service config using this parameter to change service defaults,
+	// or overwrite rendered information using raw OpenStack config format. The content gets added to
+	// to /etc/<service>/<service>.conf.d directory as custom.conf file.
+	CustomServiceConfig string `json:"customServiceConfig,omitempty"`
 }
 
 // CinderStatus defines the observed state of Cinder
 type CinderStatus struct {
-	// DbSyncHash db sync hash
-	DbSyncHash string `json:"dbSyncHash"`
-	// API endpoint
-	APIEndpoint string `json:"apiEndpoint"`
```

Comment:
could think if we just mirror the APIEndpoint from the api CR so that we have a single CR to check for details

Reply:
Ack, will do this

Reply:
Done